artemis/modules/nuclei.py

```diff
@@ -164,6 +166,38 @@ def __init__(self, *args: Any, **kwargs: Any):
             for line in open(Config.Modules.Nuclei.NUCLEI_TEMPLATES_TO_SKIP_PROBABILISTICALLY_FILE):
                 self._nuclei_templates_or_workflows_to_skip_probabilistically_set.add(line.strip())
 
+    def _run_tech_detection(self, urls: List[str]) -> Any:
+        """
+        Run technology detection on a list of URLs using Wappalyzer.
+        """
+        wappalyzer_path = os.path.join(os.path.dirname(__file__), "data", "wappalyzer")
+        main_go_path = os.path.join(wappalyzer_path, "main.go")
+        if not os.path.exists(main_go_path):
+            self.log.error(f"Wappalyzer main.go not found at {main_go_path}")
+            return {url: [] for url in urls}
+
+        try:
+            # Update the Wappalyzer package once
+            subprocess.run(["go", "get", "-u", "./..."], cwd=wappalyzer_path, check=True, capture_output=True)
+
+            temp_file_name = "/tmp/temp_urls.txt"
+            with open(temp_file_name, "w") as f:
+                for url in urls:
+                    f.write(url + "\n")
+
+            wappalyzer_output = subprocess.check_output(
+                ["go", "run", main_go_path, temp_file_name], cwd=wappalyzer_path
+            )
+            # The output is a mapping from URL to a list of detected app names
+            if os.path.exists(temp_file_name):
+                os.remove(temp_file_name)
+            return json.loads(wappalyzer_output)
+        except (subprocess.CalledProcessError, FileNotFoundError, json.JSONDecodeError) as e:
+            self.log.error(f"Error running technology detection: {e}")
+            if os.path.exists(temp_file_name):
```

Comment:
you can extract into the finally: block and therefore you won't have to repeat yoursef ;)